.gitignore

```diff
@@ -12,9 +12,9 @@ b~*.ad?
 testsuite/get_entities
 
 # Generated files on windows.
-build/
-dist/mcode/build/
-dist/mcode/lib/
+./build/
+./dist/mcode/build/
+./dist/mcode/lib/
 
```

Comment:
General note: you should split large patches like this in several smaller one (or multiple commit).  That makes reviewing much easier.

This change to .gitignore is ok, but somewhat unrelated to the other changes.

Reply:
I tried to do it. Indeed, this PR includes minimum changes to use multistage dockerfiles. I kept every unrelated enhancement in #477 as WIP.

This change precisely is motivated because some of the files in `dist/docker/build` or `dist/docker/run` were filtered by `.gitignore` if `./` is not added. I know, it makes little sense to me too.